WooCommerce/Classes/ViewRelated/Dashboard/Factories/DashboardUIFactory.swift

```diff
@@ -19,11 +21,73 @@ protocol DashboardUI: UIViewController {
 }
 
 final class DashboardUIFactory {
-    static func dashboardUI() -> DashboardUI {
-        if FeatureFlag.stats.enabled {
-            return StoreStatsAndTopPerformersViewController(nibName: nil, bundle: nil)
+    private let siteID: Int
+    private let stores: StoresManager
+    private var stateCoordinator: StatsVersionStateCoordinator?
```

Comment:
should this reference to the stateCoordinator be weak?

Reply:
yea, it can be weak. I don't think there's a retain cycle because of the `[weak self]` usage (also checked the memory debugger after pulling down to refresh and switching store), but will update to `weak` in any case

Reply:
actually, after another pass of testing I noticed the `StatsVersionStateCoordinator` was deallocated after `loadLastShownVersionAndCheckV4Eligibility` call and thus `self` is nil in the callback. I removed the `weak`, and verified again that there's no retain cycle in memory debugger